dlt/extract/incremental.py

```diff
@@ -0,0 +1,133 @@
+from hashlib import sha1
+from typing import Generic, TypeVar, Any, Optional, Callable, Protocol, List, TypedDict, cast, Union, Tuple, get_origin, Dict
+import inspect
+
+from functools import wraps
+
+from dlt.common.json import json
+from dlt.common.typing import TDataItem, AnyFun, TFun
+from dlt.common.configuration import configspec, known_sections, with_config, resolve_configuration
+from dlt.common.configuration.specs import BaseConfiguration
+from dlt.common.configuration.specs.exceptions import NativeValueError
+from dlt.common.configuration.accessors import config
+
+
+TCursorValue = TypeVar("TCursorValue", bound=Any)
```

Comment:
you probably want to bind as `str` here?

Reply:
I would expect `int` also for now. Have https://github.com/dlt-hub/dlt/issues/180 in mind too.   
Technically anything that can serialize and supports "lessthan" comparison would work.

Reply:
ahh this is `value` not `key` my bad